doc/source/main-documentation/qibolab.rst

```diff
@@ -790,3 +790,93 @@ Compatible and tested with:
+    - ``generate_model_config``: Generates and returns a multi-qubit model configuration with the specified parameters.
+    - ``generate_model_config_oneQ``: Generates and returns a one-qubit model configuration with the specified parameters.
+
+- :py:mod:`qibolab.instruments.models.methods` module: Provides common functionalities for all other model modules.
+    - ``load_model_params``: Loads a yaml file storing the model parameters into a dictionary.
+    - ``default_noflux_platform2simulator_channels``: Constructs and returns the dictionary that specifies the default mapping between platform channel names to simulator chanel names.
+
+Simulator Backends
+^^^^^^^^^^^^^^^^^^
+
+- :py:mod:`qibolab.instruments.simulator.backends.generic` module: Provides generic functionalities for different  simulator backend modules.
+    - ``dec_to_basis_string``: Converts an integer (e.g. matrix/vector index) to a generalized bitstring in the computation basis for a heterogeneous Hilbert space with a specified structure given by an ordered list of local Hilbert space dimensions.
+    - ``make_comp_basis``: Generates the computational basis states of a heterogeneous Hilbert space with a specified structure given by a list of qubit indices and their corresponding local Hilbert space dimensions.
+    - ``op_from_instruction``: Converts an instruction tuple into a quantum operator.
+        - ``process_op``: Helper function that implements only the specified type of operation between adjacent quantum objects when processing the instruction tuple.
+
+- :py:mod:`qibolab.instruments.simulator.backends.qutip_backend` module: This module provides a backend for Quantum Toolbox in Python (QuTiP) to simulate quantum devices.
+    - ``get_default_qutip_sim_opts``: Generates a default QuTiP simulation options object.
+    - :class:`qibolab.instruments.simulator.backends.qutip_backend.Qutip_Simulator`: This class builds pulse simulator components using the QuTiP backend. The pulse simulation is implemented by the `qevolve` function, which is called by the `PulseSimulator.play` method with `channel_waveforms` (:class:`dict`) as input. The `Qutip_Simulator` class does not directly interact with Qibolab. Instead, the `channel_waveforms` dictionary is generated from the `qibolab.pulses.PulseSequence` within the :class:`qibolab.instruments.pulse_simulator.PulseSimulator` class, which uses Qibolab primitives.
+    - ``make_arbitrary_state``: Creates a quantum state object using the given state data and dimensions.
+    - ``function_from_array``: Returns function given a output array y and an input array x.
+    - ``extend_op_dim``: general helper function to extend the dimension of the input operator from the local Hilbert space (typically involving 1 or 2 qubits) to a larger Hilbert space.
```

Comment:
This content is describing module-by-module, class-by-class, and function-by-function the additions in this PR.

However, this would be rendered here:
https://qibo.science/qibolab/stable/main-documentation/qibolab.html
where instead only the high-level description should be provided (ok, maybe there are already more than needed even in the other sections, but that's not a good reason to keep going in the wrong direction...).

This is not needed here, because an equivalent description will be automatically extracted by Sphinx from the docstrings, and this will be found here:
https://qibo.science/qibolab/stable/api-reference/modules.html

Reply:
So I should basically provide an overview description of the emulator. Do you suggest that I follow the example from Dummy platform?

Reply:
You could add a paragraph below "Dummy platform", and describe that the emulators are also an option, that would be nice.

However, I'd also try to treat the `PulseSimulator` as an `Instrument`.
It has to be marked explicitly, since it is not corresponding to any physical instrument.
But from the point of view of the execution and platform creation, it should be possible to use it as any other instrument.

Reply:
Fixed in c68c3c5f23da65dde887fb6ec169c866f126e203